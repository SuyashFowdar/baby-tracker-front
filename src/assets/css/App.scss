header {
  box-shadow: 0 0 10px inset rgba(0, 0, 0, 0.7);
  height: 50px;
  position: fixed;
  width: 100vw;
  bottom: 0;
  background-color: rgb(49, 57, 68);

  .link {
    font-size: 20px;
    color: rgb(174, 177, 182);

    span {
      font-size: 16px;
    }
  }
}

.main {
  height: calc(100vh - 50px);
  overflow-y: auto;
}
