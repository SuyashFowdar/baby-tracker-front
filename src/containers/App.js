import React, { useEffect, useState } from 'react';
import {
  HashRouter as Router,
  Link,
  Switch,
  Route,
} from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSignOutAlt, faHome } from '@fortawesome/free-solid-svg-icons';
import Home from './Home';
import Measure from './Measure';
import Admin from './Admin';
import Login from './Login';
import query from '../query';
import '../assets/css/App.scss';
import { setMeasures } from '../actions';

const App = () => {
  const dispatch = useDispatch();
  const [errorMessage, setErrorMessage] = useState('');
  const logout = () => {
    localStorage.clear();
    window.location.reload();
  };

  useEffect(() => {
    if (localStorage.token) {
      query('GET', 'measurements', null, (result) => {
        if (result.measures && result.measurements) {
          dispatch(setMeasures(result));
        } else {
          setErrorMessage(result.error);
        }
      });
    }
  });

  return (
    <div>
      {localStorage.token
        ? (
          <Router>
            <div className="error row main-center cross-center">{errorMessage}</div>
            <header className="row main-space-around">
              <Link to="/" className="col cross-center main-center shadow-5 link flex">
                <FontAwesomeIcon icon={faHome}>Sign Out</FontAwesomeIcon>
                <span>Home</span>
              </Link>
              <button type="button" onClick={logout} className="link col cross-center main-center flex">
                <FontAwesomeIcon icon={faSignOutAlt}>Sign Out</FontAwesomeIcon>
                <span>Log out</span>
              </button>
            </header>
            <div className="main">
              <Switch>
                <Route exact path="/"><Home /></Route>
                <Route path="/measure/:id"><Measure /></Route>
                <Route path="/admin"><Admin /></Route>
              </Switch>
            </div>
          </Router>
        )
        : <Login />}
    </div>
  );
};

export default App;
