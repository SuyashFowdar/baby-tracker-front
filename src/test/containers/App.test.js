import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import App from '../../containers/App';
import reducer from '../../reducers/index';

it('renders correctly', () => {
  const store = createStore(reducer);
  const tree = renderer
    .create(<Provider store={store}><App /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
