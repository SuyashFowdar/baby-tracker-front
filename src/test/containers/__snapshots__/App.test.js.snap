// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="login"
  >
    <div
      className="col cross-center main-center"
    >
      <form
        className="col cross-center"
        onSubmit={[Function]}
      >
        <h2>
          Sign In
        </h2>
        <input
          name="name"
          placeholder="Username"
          type="text"
        />
        <input
          name="password"
          placeholder="Password"
          type="password"
        />
        <input
          className="link w-90"
          type="submit"
          value="Sign In"
        />
        <button
          className="link button w-90"
          onClick={[Function]}
          type="button"
        >
          or create new account
        </button>
      </form>
    </div>
    <div
      className="error row main-center cross-center"
    >
      
    </div>
  </div>
</div>
`;
